File: employee-management-client/src/app/department.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class DepartmentService {

  private baseUrl = 'http://localhost:8082/';

  constructor(private http: HttpClient) {
    
   }

  addDepartment(dept: Object): Observable<Object> {
    return this.http.post(`${this.baseUrl}/department`, dept);
  }

  deleteDepartment(id: number): Observable<any> {
    return this.http.delete(`${this.baseUrl}/department/${id}`);
  }

  getDepartmentList(): Observable<any> {
    return this.http.get(`${this.baseUrl}/departments`);
  }
}

File: EmployeeManagement/src/main/java/com/socgen/EmployeeManagement/service/DepartmentService.java
package com.socgen.EmployeeManagement.service;

import java.util.List;

import com.socgen.EmployeeManagement.model.Department;

public interface DepartmentService {

	public Department addDepartment(Department dept);
	
	public void deleteDepartment(int id);
	
	public List<Department> getDepartments();
	
}
